### ISSUE TITLE
FreeBSD support for /bin/puppet_install.sh

### ISSUE THREAD

**Issue by lesinigo** [example42/control-repo#76]
BSD is currently "not supported" by the install script.

_TL;DR:_ if you want FreeBSD support in a single script for all operating systems it must be made sh-compatibile, at least for the FreeBSD code path, or require a FreeBSD system where someone or something pre-installed `bash`. You will also have to live with facter2 at least for some time or compile relevant packages from the port collections (on every node, or just on one node and distribute your own binary packages).

Here are some relevant info:
- on a fresh FreeBSD install there is no bash but you do [get the Bourne Shell](https://www.freebsd.org/doc/en/articles/linux-users/shells.html) (in addition to tcsh and csh). Bourne Again Shell 4.4 is [available through a package](https://www.freebsd.org/cgi/ports.cgi?query=%5Ebash-&stype=all&sektion=all)
- AFAIK there are no Puppet official packages for the open source edition on FreeBSD
- there is [a FreeBSD package for puppet4](https://www.freebsd.org/cgi/ports.cgi?query=puppet4&stype=all) which is usually more ore less up to date with upstream releases
- caveat: the package depends on [the facter2 package](https://www.freebsd.org/cgi/ports.cgi?query=%5Erubygem-facter&stype=name&sektion=all), a package for [Facter 3.1 is available](https://www.freebsd.org/cgi/ports.cgi?query=%5Efacter&stype=name&sektion=all) but will conflict with the former
- via FreeBSD Ports you could [configure Puppet4 with Facter3 support](https://svnweb.freebsd.org/ports/head/sysutils/puppet4/Makefile?revision=432695&view=markup#l40), but that would require (automated) build from sources
- newer Facter versions are not currently supported, [pending integration of the devel/leatherman port](https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=213624)

An sh-compatibile unattended Puppet OpenSource installation on FreeBSD using its package system (with Facter2) boils down to this (the `bootstrap` and `update` phases are required if it is a fresh install where `pkg` has not been used before):
```
#!/usr/bin/env sh
export ASSUME_ALWAYS_YES=YES
pkg bootstrap
pkg update
pkg install puppet4
```
This will install the agent and the master (configured to run in mongrel) with the appropriate rc scripts to start them at boot (disabled by default).

We are using a variant of that script to set up production FreeBSD-11-RELEASE systems and as far as I can tell they integrate without problems with a Puppet Collections 1 server running on linux, but we are also considering switching to building it through ports to get the newer facter3.

**Comment by alvagante** [comment]
Thanks for the info, will remain as reference for who wants/needs to implement it via a PR (eventually using sh instead of bash, as long as nothing breaks).
Or I will do it as when possible.
